File: src/types/membership.ts
export type MembershipStatus = "ACTIVE" | "EXPIRED" | "FROZEN" | "CANCELED";

// Matches CreateMembershipCommand (POST /api/memberships)
export interface MembershipRequest {
  memberId: number;
  packageId: number;
  startDate: string;
  price?: number;
  debt?: number;
  description?: string;
}

// Matches MembershipDto (GET /api/memberships/{id}, GET /api/memberships/by-member/{memberId})
export interface Membership {
  id: number;
  memberId: number;
  memberFullName: string;
  packageId: number;
  packageName: string;
  status: MembershipStatus | string;
  startDate: string;
  endDate: string;
  price: number;
  debt?: number;
  description?: string;
}

export interface MembershipHistory {
  id: number;
  price: number;
  debt?: number;
  packageName: string;
  description?: string;
  timestamp: string;
}

File: backend/src/FitMaster.Domain/Entities/Memberships/Membership.cs
using FitMaster.Domain.Common;
using FitMaster.Domain.Entities.Identity;
using FitMaster.Domain.Enums;

namespace FitMaster.Domain.Entities.Memberships;

public class Membership : BaseEntity
{
    public required long MemberId { get; set; }

    public User Member { get; set; } = null!;

    public required long PackageId { get; set; }

    public Package Package { get; set; } = null!;

    public MembershipStatus Status { get; set; } = MembershipStatus.Active;

    public required DateOnly StartDate { get; set; }

    public required DateOnly EndDate { get; set; }

    public decimal Price { get; set; }

    /// <summary>Outstanding amount the member still owes, if any.</summary>
    public decimal? Debt { get; set; }

    public string? Description { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public ICollection<Revenue> Revenues { get; set; } = new List<Revenue>();
}
